Matches/Regex.swift

```diff
@@ -0,0 +1,67 @@
+//  Copyright © 2018 Inloop, s.r.o. All rights reserved.
+
+import Foundation
+
+public struct Regex: Match {
+    let pattern: String
+    let options: NSRegularExpression.Options
+
+    public init(pattern: String, options: NSRegularExpression.Options = .caseInsensitive) {
+        self.pattern = pattern
+        self.options = options
+    }
+
+    public static func contains(_ pattern: String, ignoringCase: Bool = true) -> Regex {
+        return Regex(pattern: pattern, options: ignoringCase ? .caseInsensitive : [])
+    }
+
+    /// \bstring\b
+    public static func wordMatches(string: String, ignoringCase: Bool = true) -> Regex {
+        return Regex(pattern: "\\b\(string)\\b", options: ignoringCase ? .caseInsensitive : [])
+    }
+
+    /// \bstring
+    public static func wordStarts(with string: String, ignoringCase: Bool = true) -> Regex {
+        return Regex(pattern: "\\b\(string)", options: ignoringCase ? .caseInsensitive : [])
+    }
+
+    /// string\b
+    public static func wordEnds(with string: String, ignoringCase: Bool = true) -> Regex {
+        return Regex(pattern: "\(string)\\b", options: ignoringCase ? .caseInsensitive : [] )
+    }
+
+    /// (-?\d{1,3}((,|\.)?\d{3})*((,|\.)\d*?))(\D|\$)
+    public static let currency: Regex = {
+        return Regex(pattern: "(-?\\d{1,3}((,|\\.)?\\d{3})*((,|\\.)\\d*?))(\\D|\\$)")
+    }()
+
+    /// \s(\w+)(?>\.|!|\?)*$
+    public static let lastWord: Regex = {
+        return Regex(pattern: "\\s(\\w+)(?>\\.|!|\\?)*$", options: [.caseInsensitive, .anchorsMatchLines])
+    }()
+
+    /// ^\w+
+    public static let firstWord: Regex = {
+        return Regex(pattern: "^\\w+", options: [.caseInsensitive, .anchorsMatchLines])
+    }()
+
+    public func ranges(in base: String) -> [NSRange] {
+        do {
+            let regex = try NSRegularExpression(pattern: pattern, options: options)
+            let matches = regex.matches(in: base, options: [], range: NSRange(base.startIndex..., in: base))
+            var ranges: [NSRange] = []
+            for match in matches {
+                if match.numberOfRanges > 1 {
+                    for index in 1..<match.numberOfRanges {
```

Comment:
why since 1 ?

Reply:
because range 0 is whole match. not needed when matching capture groups.